### DISCUSSION TITLE
How Are Coding Collisions without Punctuation Handled?

### DISCUSSION THREAD

**Discussion by TheCedarPrince** [JuliaHealth/ICD_GEMs.jl#10] [Q&A]
Hi @InterdisciplinaryPhysicsTeam team,

Question for you all regarding #9 .
Given that punctuation in codes is not supported or valid in your implementation, how do you disambiguate codes that look like this:

```
E11.4
E11.40
```

when `E11.4` means "[Type 2 diabetes mellitus with neurological complications](https://athena.ohdsi.org/search-terms/terms/1567965)" and `E11.40`refers to "[Type 2 diabetes mellitus with diabetic neuropathy, unspecified](https://athena.ohdsi.org/search-terms/terms/45605403)"?
Do you preserve zeros and then map the 4 digit code to the 4 digit billing code and the 5 digit code to the 5 digit billing code?
The reason I was wondering is I have done analysis incorrectly in the past when I remove punctuation like this and accidentally map to different codes - was curious how you handled this issue as I had to resort to a lot of manual manipulation.

**Comment by InterdisciplinaryPhysicsTeam** [comment]
Hi @TheCedarPrince,

Thanks for this discussion!

Since this package has been originally thought as a translation utility, it only deals with codes (like `E1140`) and not directly with (sub)categories (like `E114`). After all, (sub)categories are not reported in the CDC's GEM files as single entries (they only contain codes), so we don't have a hard-coded correspondence between ICD-10 (sub)categories and ICD-9 (sub)categories and vice-versa. 

Anyway you may get the "translation of a (sub)category" by specifying it as a range in the conversion utility:

```julia
using ICD_GEMs

subcategory_ICD9 = execute_applied_mapping(GEM_I10_I9_dictionary, ["E114-E114"])
```

The signature `"E114-E114"` means "all codes from the first that starts with `E114` to the last that ends with `E114`". Notice that you can have arbitrary precision (i.e. number of digits) [on both ends](https://inphyt.github.io/ICD_GEMs.jl/stable/#Tutorial). To see all `ICD-10` codes that are included in the subcategory `E144`, you may refer to the function:

```julia
get_ICD_code_range("E114-E114", "ICD-10")
```

```bash
6-element Vector{String}:
 "E1140"
 "E1141"
 "E1142"
 "E1143"
 "E1144"
 "E1149"
```

More details in the [docs](https://inphyt.github.io/ICD_GEMs.jl/stable/internals/#ICD_GEMs.get_ICD_code_range-Tuple{String,%20String}) and in the [GEMs official documentation](https://github.com/InPhyT/ICD_GEMs.jl/tree/main/data).

If you think you'd benefit from an extension of this package, please consider submitting a PR or filing a [feature request](https://github.com/InPhyT/ICD_GEMs.jl/issues/new/choose). 

CC: @pitmonticone @ClaudMor
